contracts/dapis/proxies/ProxyFactory.sol

```diff
@@ -0,0 +1,118 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "./DataFeedProxy.sol";
+import "./DapiProxy.sol";
+import "./DataFeedProxyWithOev.sol";
+import "./DapiProxyWithOev.sol";
+
+contract ProxyFactory {
+    event DeployedDataFeedProxy(
+        address indexed proxyAddress,
+        bytes32 indexed dataFeedId,
+        bytes metadata
+    );
+
+    event DeployedDapiProxy(
+        address indexed proxyAddress,
+        bytes32 indexed dapiName,
+        bytes metadata
+    );
+
+    event DeployedDataFeedProxyWithOev(
+        address indexed proxyAddress,
+        bytes32 indexed dataFeedId,
+        address indexed oevBeneficiary,
+        bytes metadata
+    );
+
+    event DeployedDapiProxyWithOev(
+        address indexed proxyAddress,
+        bytes32 indexed dapiName,
+        address indexed oevBeneficiary,
+        bytes metadata
+    );
+
+    address public immutable dapiServer;
+
+    constructor(address _dapiServer) {
+        require(_dapiServer != address(0), "dAPI server zero");
+        dapiServer = _dapiServer;
+    }
+
+    function deployDataFeedProxy(bytes32 dataFeedId, bytes calldata metadata)
+        external
+        returns (address proxyAddress)
+    {
+        bytes memory initcode = abi.encodePacked(
+            type(DataFeedProxy).creationCode,
+            abi.encode(dapiServer, dataFeedId)
+        );
+        // solhint-disable-next-line no-inline-assembly
+        assembly {
+            proxyAddress := create2(0, add(initcode, 0x20), mload(initcode), 0)
```

Comment:
I initially thought the reason deterministic deployments being difficult/impossible on some chains was due to difference in `CREATE2` support/behavior so it should be avoided, but I couldn't find anything about that when I searched for it specifically. So I ended up using it.